### PR TITLE
feat: add complexity-aware routing to lfg/slfg

### PR DESCRIPTION
**Pull request by tmchow**
## Summary

Stacked on #346 (pipeline orchestration + pipeline mode). That PR made lfg/slfg always run the full brainstorm-plan-work-review-test pipeline. This PR adds a routing layer so the pipeline is **right-sized to the task** which allows `lfg`/`slfg` to always be used as a by default without worrying about being buried in overweight process.

### The problem

`/lfg fix the typo on line 42 of foo.ts` should not generate a plan file, run brainstorm, invoke multi-agent review, and record a feature video. The full pipeline is the right tool for complex features but massive overkill for trivial fixes.

### The solution

Both lfg and slfg now have a **Phase 0: Assess and Route** that classifies the task into one of three execution paths:

| Tier | When | What runs |
|------|------|-----------|
| **Direct** | Obvious, self-contained fix (typo, rename, config change) | Make the edit, verify (typecheck/lint/test), done. No skills loaded. |
| **Lightweight** | Clear, bounded task with specified requirements | Do the work directly with verification and self-review. Skips brainstorm, plan, and multi-agent review. |
| **Full Pipeline** | Scope, ambiguity, or risk warrants structured planning | Complete skill chain: brainstorm -> plan -> work -> review -> test -> video |

**Routing bias:** under-route when uncertain. A typo fix that accidentally gets Lightweight just adds a self-review. A complex feature that accidentally gets Lightweight will have the user ask for more. Both are cheaper than running a 10-step pipeline for a one-line fix.

### Design decisions

- **No structured classification dimensions.** Citadel's `/do` skill uses 6 explicit axes (SCOPE, COMPLEXITY, INTENT, etc.) plus regex pattern matching. We tested whether the agent needs that structure -- it doesn't. Natural language tier descriptions + bias rules are sufficient. Validated with 16 eval prompts across 2 rounds (trivial fixes, mechanical renames, bounded features, ambiguous tasks, high-risk domains) with 16/16 correct routing decisions.
- **Agent intelligence over rigid rules.** The routing doesn't keyword-match ("payments" doesn't auto-route to Full Pipeline). The agent reasons about actual task risk -- deleting a deprecated PaymentV1 module routes Lightweight (mechanical deletion), while adding a Stripe webhook routes Full Pipeline (external API + unspecified business logic).
- **Transparent announcements.** The agent announces its routing decision and reasoning in one line before executing, so the user can see the classification and intervene if wrong.

## Additional changes

- **Lightweight verification gap fixed** (P1 from review): Lightweight now requires "verify it works (typecheck, lint, or test if applicable)" matching Direct's bar -- previously only had "quick self-review"
- **Feature-video completion gate** (P2 from review): Step 10 now conditionally gates on video being in PR only when feature-video actually ran, rather than unconditionally or not at all
- **README updates**: Both READMEs updated to describe routing behavior with tier examples instead of claiming lfg/slfg always run the full pipeline
- **Solution doc updated**: `docs/solutions/` learning doc corrected to reflect routing (previously said "brainstorm is always invoked")

## Test plan

- [x] 16 eval prompts across 2 rounds testing routing classification
  - Round 1 (clean cases): typo fix, rename, loading spinner, input validation, dark mode, OAuth2 redesign, specific API endpoint, vague perf improvement -- 8/8 correct
  - Round 2 (edge cases): rate limiting, copyright year, add logging, remove deprecated module, add tests, DB column rename, Stripe webhook, flaky test fix -- 8/8 correct
- [x] Interactive `/lfg` session with a trivial task to verify Direct path executes correctly
- [x] Interactive `/lfg` session with a complex task to verify Full Pipeline still works end-to-end
- [x] `bun run release:validate` passes

### CODE REVIEW

**Comment by tmchow** [comment]
@codex review again

File: plugins/compound-engineering/skills/lfg/SKILL.md [hunk 1]
```diff
@@ -1,11 +1,52 @@
...
+
+---
+
+### Direct
+
+The fix is obvious and self-contained. No planning or multi-agent review needed.
+
+Before changing files, preserve the same branch/worktree safety as `ce:work` Phase 1: choose the right branch first, and never commit directly to the default branch without explicit user permission.
+
+Make the change, verify it works (typecheck, lint, or test if applicable), then preserve the same wrap-up contract as `ce:work` Phase 4: commit it, push it, and create or update the PR before outputting `<promise>DONE</promise>`.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply plugins/compound-engineering/skills/lfg/SKILL.md]: Fixed in `cb7365d`.

Direct and Lightweight now explicitly preserve the applicable `ce:work` Phase 4 ship requirements instead of reducing them to commit/push/PR only. The fast paths now call out:
- PR creation/update
- required `## Post-Deploy Monitoring & Validation`
- screenshots plus uploaded image URLs for browser UI changes

That change was applied to both `lfg` and `slfg`.
